src/GraphQL/Utilities/DirectiveVisitorSelector.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using GraphQL.Types;
+using GraphQLParser.AST;
+
+namespace GraphQL.Utilities
+{
+    public class DirectiveVisitorSelector : IVisitorSelector
+    {
+        private IDictionary<string, Type> _directiveVisitors;
+        private readonly Func<Type, SchemaDirectiveVisitor> _typeResolver;
+
+        public DirectiveVisitorSelector(
+            IDictionary<string, Type> directiveVisitors,
+            Func<Type, SchemaDirectiveVisitor> typeResolver)
+        {
+            _directiveVisitors = directiveVisitors;
+            _typeResolver = typeResolver;
+        }
+
+        public IEnumerable<ISchemaNodeVisitor> Select(object node)
```

Comment:
These 3 (may be more in future) branches of the code can be reduced to just one, if you create an interface like
```c#
public interface IHasDirectivesDefinition
{
  public IEnumerable<GraphQLDirective> Directives { get; set; }
}
```
in GraphQLParser project and apply it to `GraphQLObjectTypeDefinition`, `GraphQLFieldDefinition`, `GraphQLEnumValueDefinition` classes:

```c#
 public IEnumerable<ISchemaNodeVisitor> Select(object node)
 {
          
                var ast = obj.GetAstType<IHasDirectivesDefinition>();
                if (ast == null) yield break;
                foreach (var visitor in BuildVisitors(ast.Directives))
                {
                    yield return visitor;
                }
}    
```

It will be easier and more flexible.

Reply:
Yes, agreed.  Something like that is needed in the Parser.

Reply:
I can do it. Ok?

Reply:
That would be great, thanks!

Reply:
https://github.com/graphql-dotnet/parser/pull/34

There are more changes than are required for this task, but they are fairly simple.